src/dlr_common.cc

```diff
@@ -68,26 +84,30 @@ void dlr::LoadJsonFromFile(const std::string& path, nlohmann::json& jsonObject)
   }
 }
 
-DLRBackend dlr::GetBackend(std::vector<std::string> dir_paths) {
-  // Support the case where user provides full path to hexagon file.
-  if (EndsWith(dir_paths[0], "_hexagon_model.so")) {
-    return DLRBackend::kHEXAGON;
+std::vector<std::string> dlr::FindFiles(std::vector<std::string> paths) {
+  std::vector<std::string> files;
+  for (auto path : paths) {
+    dlr::ListDir(path, files);
   }
-  // Scan Directory content to guess the backend.
-  std::vector<std::string> paths;
-  for (auto dir : dir_paths) {
-    dlr::ListDir(dir, paths);
-  }
-  for (auto filename : paths) {
+  return files;
+}
+
+DLRBackend dlr::GetBackend(const std::vector<std::string>& files) {
+  // Scan files to guess the backend.
+  bool hasSO = false;
```

Comment:
Rename to `has_so` or `has_lib` to follow naming convention

Reply:
done